scripts/Impurity-example.jl

```diff
@@ -0,0 +1,198 @@
+# We save the Monte Carlo data every 1000 steps.
+# Progress messages are suppressed with `io=devnull`, for a real job one should remove the 
+# line to invoke the default `io` and reenable the output messages.
+r_strat = ReportToFile(
+    save_if = is_mpi_root(),
+    filename = "mpi_df_$(η)_$(P).arrow",
+    chunk_size = 1000,
+    return_df = true, # change it to `false` when running the real job
+    io=devnull # remove this line when running the real job
+    )
+
+# We will turn on the replica, but without operators for a fast equilibration.
+el2 = @elapsed df, state = lomc!(ham,dv; params, s_strat, r_strat, replica = AllOverlaps(2, nothing), laststep = (steps_equilibrate+steps_warmup))
+@mpi_root @info "Replica fciqmc completed in $(el2) seconds."
+
+# ## Stage 3: Running with the real Hamiltonian with replica and observables
+
+# We now at the last stage of the calculation, doing replica FCIQMC with a serious of 
+# G2 correlators with distance `d` from `0` to `m`. See [`G2Correlator`](@ref).
+# Here we save data every 1000 steps, but using a smaller `chunk_size` like 10 or even 1
+# is highly recommended as replica FCIQMC with many observables being calculated are very 
+# expensive and you don't want to loose too much data if the job stops before finishes.
+# Progress messages are suppressed with `io=devnull`, for a real job one should remove the 
+# line to invoke the default `io` and reenable the output messages.
+r_strat = ReportToFile(
+    save_if = is_mpi_root(),
+    filename = "mpi_df_g2_$(η)_$(P).arrow",
+    chunk_size = 1000,
+    return_df = true, # change it to `false` when running the real job
+    io = devnull # remove this line when running the real job
+    )
+
+# Setting up a tuple of G2 correlators:
+g = Tuple(G2Correlator.(0:m))
+# By default, for a two-component system the cross-component G2 operators are set up.
+# If you want to calculate the correlations within a single component, `G2Correlator(d,:first)` 
+# or `G2Correlator(d,:second)` can be called based on your choice.
+
+# Carry over information from the previous stage and set up a new `QMCState`:
+new_state = Rimu.QMCState(
+    state.hamiltonian, state.replicas, Ref(Int(state.maxlength)),
+    state.m_strat, r_strat, state.s_strat, state.τ_strat, state.threading, state.post_step, AllOverlaps(2, g)
+    )
+# The final stage 
+el3 = @elapsed df2, state2 = lomc!(new_state; laststep = (steps_equilibrate+steps_warmup+steps_final))
+@mpi_root @info "Replica fciqmc with G2 completed in $(el3) seconds."
+println("MPI run finished!")
+
+# ## Post-calculation analysis
+
+# Typically, one should not include any analyses when using MPI, as they will be calculated multiple
+# time unless you put the `@mpi_root` macro everywhere. Even so, all other MPI ranks apart from the root
+# will be idling and wasting CPU hours on a HPC unit.
+# But here, let's have a look of the calculated G2 correlations:
```

Comment:
This part of the example overlaps with the third example script, i.e. just listing and checking the G2 values. Perhaps this is all that can be done in a small example script, but can something be said about the effect of the impurity?

Reply:
Yes, it looks like the analysis that follows is already shown and explained in another example script. Maybe then it would be better to just describe in words what kind of analysis can be done on the data that has been saved to disk and refer to Example 3.